src/DataProtection/shared/src/TrimmerWarning.cs

```diff
@@ -0,0 +1,9 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.DataProtection;
+
+internal static class TrimmerWarning
+{
+    public const string Message = "Data Protection may attempt to load types in a way that cannot be statically analyzed. Ensure all required types are preserved.";
```

Comment:
I think we need to improve this error message. "Ensure all required types are preserved." is what they need to do, but it doesn't give any indication about how they do it.

The simplest way to provide more info is to link to a doc. I took a look at https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-self-contained and there isn't any content about what to do here.

Reply:
We could add trimming specific guidance to the [DataProtection docs](https://docs.microsoft.com/aspnet/core/security/data-protection/introduction?view=aspnetcore-6.0). In theory, the average user wouldn't run in to trimming specific issues with this library - it's only when they start configuring providers via xml configs or the registry that it runs the risk of being trimmed away.

Reply:
I agree.

This doesn't need to block this PR. Trimming docs issue: https://github.com/dotnet/aspnetcore/issues/41229